src/transformer/descriptor/enum/enumDeclaration.ts

```diff
@@ -1,12 +1,26 @@
 import * as ts from 'typescript';
 import { TypeChecker } from '../../typeChecker/typeChecker';
+import { RandomPropertyAccessor } from '../random/random';
+import { IsTsAutoMockRandomEnabled } from '../../../options/random';
 
 export function GetEnumDeclarationDescriptor(node: ts.EnumDeclaration): ts.Expression {
   const typeChecker: ts.TypeChecker = TypeChecker();
-  const type: ts.LiteralType = typeChecker.getTypeAtLocation(node.members[0]) as ts.LiteralType;
+  const typesList: ts.LiteralType[] = node.members.map((it: ts.EnumMember) => typeChecker.getTypeAtLocation(it)) as ts.LiteralType[];
 
-  if (type.hasOwnProperty('value')) {
-    return ts.createLiteral(type.value);
+  if (IsTsAutoMockRandomEnabled()) {
+    const nodesList: ts.Expression[] = typesList.map((type: ts.LiteralType) => {
+      if (type.hasOwnProperty('value')) {
+        return ts.createLiteral(type.value);
+      }
+
+      return ts.createLiteral(0);
```

Comment:
I think we can use the index number here :) Instead of 0 we can use the index from the array map

```ts
const nodesList: ts.Expression[] = typesList.map((type: ts.LiteralType, index: number) => {
      if (type.hasOwnProperty('value')) {
        return ts.createLiteral(type.value);
      }

      return ts.createLiteral(index); // What is this scenario??? read below
```

We should add a test for random values for enum that do not have a value. WHen this is happening? It's happening for enum in module declaration. You can find a test for non random in test/transformer/descriptor/enum.test.ts. This is the test that we should have also for random enum

```ts
describe('for enum from a module', () => {
    it('should assign the first enum value', () => {
        const properties: NameSpace.WithEnum = createMock<NameSpace.WithEnum>();

        // We cannot check against the real ENUM because is a d.ts, it cannot be used as a value
        expect(properties.propertyWithoutDefaultValue).toEqual(0);
        expect(properties.propertyWithDefaultValue).toEqual('1');
    });
});
```